app/gui/view/graph-editor/src/lib.rs

```diff
@@ -1539,6 +1622,69 @@ impl GraphEditorModel {
 }
 
 
+// === Add node ===
+impl GraphEditorModel {
+    pub fn add_node(&self) -> NodeId {
+        self.frp.add_node.emit(());
+        let (node_id, _) = self.frp.node_added.value();
+        node_id
+    }
+
+    pub fn add_node_below(&self, above: NodeId) -> NodeId {
+        let pos = self.find_free_place_under(above);
+        self.add_node_at(pos)
+    }
+
+    pub fn add_node_at(&self, pos: Vector2) -> NodeId {
+        let node_id = self.add_node();
+        self.frp.set_node_position((node_id, pos));
+        node_id
+    }
+
+    pub fn find_free_place_under(&self, node_above: NodeId) -> Vector2 {
+        let above_pos = self.node_position(node_above);
+        let y_gap = self.frp.default_y_gap_between_nodes.value();
+        let y_offset = y_gap + node::HEIGHT;
+        let starting_point = above_pos - Vector2(0.0, y_offset);
+        let direction = Vector2(1.0, 0.0);
```

Comment:
I think it should mean "left", but `(1,0)` means right – are we not searching place on the left side of currently occupied place?

Reply:
I just wanted to keep it as it was. But of course I can change to left.